### PR TITLE
Adding support for dockerlint json files

### PR DESCRIPTION
**Pull request by amandel**
Support for json format output of https://github.com/projectatomic/dockerfile_lint

### CODE REVIEW

File: src/main/java/edu/hm/hafner/analysis/parser/DockerLintParser.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        return builder.build();
+    }
+
+    private String colapseReferenceUrl(final Object refUrl) {
+        StringBuilder referenceUrl = new StringBuilder();
+        if (refUrl instanceof JSONArray) {
+            for (Object part : (JSONArray) refUrl) {
+                referenceUrl.append(part);
+            }
+        } else {
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/DockerLintParser.java:91]:
```suggestion
        } 
        else {
```

**Comment by uhafner** [comment]
The new parsers are now part of the 8.2.0 release: https://plugins.jenkins.io/analysis-model-api/.

If you like, you can now register the parser in the warnings plugin, here is an [example](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/analysis/warnings/EsLint.java). There you can define an icon, a help text, default pattern, etc.

**Comment by amandel** [comment]
Thank you. The sample helped: https://github.com/jenkinsci/warnings-ng-plugin/pull/551

**Comment by wwuck** [comment]
@amandel The github repository link for dockerfile_lint appears to be dead.  I get a 404 error when visiting the link.  Has it moved somewhere else?

**Comment by amandel** [comment]
It was active some days ago. I'm not aware of a new location. :( https://hub.docker.com/r/projectatomic/dockerfile-lint does still exist. Let's hope that it is just temporarily.

**Comment by wwuck** [comment]
https://lists.projectatomic.io/projectatomic-archives/atomic/2019-February/msg00001.html
http://www.projectatomic.io/blog/2019/11/fedora-atomic-host-nearing-eol/

Do those messages/posts indicate that this dockerfile_lint tool has been abandoned upstream, or are they unrelated to the linting tool?

I can still see the dockerfile_lint repository in google search results at the top of the list (https://www.google.com/search?q=projectatomic+dockerfile_lint) but clicking on the link still takes me to a 404 page :(

**Comment by wwuck** [comment]
Seems to be up again! https://github.com/projectatomic/dockerfile_lint

**Pull request merged**